tfhe/src/c_api/utils.rs

```diff
@@ -25,22 +25,39 @@ pub fn check_ptr_is_non_null_and_aligned<T>(ptr: *const T) -> Result<(), String>
     Ok(())
 }
 
-pub fn get_mut_checked<'a, T>(ptr: *mut T) -> Result<&'a mut T, String> {
+/// Get a mutable reference from a pointer checking the pointer is well aligned for the given type.
+///
+/// # Safety
+///
+/// Caller of this function needs to make sure the aliasing rules for mutable reference are
+/// respected.
+///
+/// The basics are: at any time only a single mutable reference may exist to a given memory location
+/// XOR any number of immutable reference may exist to a given memory location.
+pub(super) unsafe fn get_mut_checked<'a, T>(ptr: *mut T) -> Result<&'a mut T, String> {
```

Comment:
Also, the pointer has to be valid pointer to a variable of type T, the non-null and alignment checks are small safeguards, e.g if the pointer actually points to some `V` that has the same alignement requirement as a `T` then it would still pass. 

Is that correct ?

Reply:
Correct, I will add to the safety

Reply:
If the pointer provided on the C side is wrong we will have a pointer to a type we are not expecting